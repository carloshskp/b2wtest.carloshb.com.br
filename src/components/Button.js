import React from 'react';
import PropTypes from 'prop-types';

const Button  = ({ text, action }) => <button onClick={action}>
    { text }
</button>;

Button.propTypes = {
    text: PropTypes.string.isRequired,
    action: PropTypes.func.isRequired
};

export default Button;