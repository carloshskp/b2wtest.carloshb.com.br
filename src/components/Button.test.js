import React from 'react';
import renderer from 'react-test-renderer';

import Button from './Button';

it('renders Button component', () => {
    const box = renderer.create(
        <Button
            text='NEXT'
            action={() => {}}
        />
    );

    let tree = box.toJSON();

    expect(tree).toMatchSnapshot();
});
