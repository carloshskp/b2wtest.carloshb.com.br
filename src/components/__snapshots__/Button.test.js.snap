// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Button component 1`] = `
<button
  onClick={[Function]}
>
  NEXT
</button>
`;
